include/revng/Support/TemporaryFile.h

```diff
@@ -14,20 +14,47 @@ class TemporaryFile {
 private:
   llvm::SmallString<32> Path;
 
+  TemporaryFile(const llvm::StringRef Path, int /* dummy */) :
+    Path(Path.str()) {
+    llvm::sys::RemoveFileOnSignal(Path);
+  }
+
 public:
-  TemporaryFile(const llvm::Twine &Prefix, llvm::StringRef Suffix = "") {
+  TemporaryFile(const llvm::Twine &Prefix, const llvm::StringRef Suffix = "") {
```

Comment:
`const` on non-reference arguments doesn't make much sense. Maybe you meant `const &` but `StringRef`s are meant to be passed by copy (like pointers).

Reply:
Dropped all `const llvm::(Array|String)Ref`s